AutocompleteClient/FW/Logic/Worker/ConstructorIO.swift

```diff
@@ -80,12 +90,18 @@ public class ConstructorIO: AbstractConstructorDataSource {
     }
     
     private func buildRequest(fromTracker tracker: CIOConversionTrackData) -> URLRequest {
-        let requestBuilder = TrackConversionRequestBuilder(tracker: tracker, autocompleteKey: self.autocompleteKey)
+        var trackData: HasSectionName = tracker
+        self.attachDefaultSectionNameIfNeeded(&trackData)
+        
+        let requestBuilder = TrackConversionRequestBuilder(tracker: trackData as! CIOConversionTrackData, autocompleteKey: self.autocompleteKey)
         return requestBuilder.getRequest()
     }
 
     private func buildRequest(fromTracker tracker: CIOAutocompleteClickTrackData) -> URLRequest {
-        let requestBuilder = TrackAutocompleteClickRequestBuilder(tracker: tracker, autocompleteKey: self.autocompleteKey)
+        var trackData: HasSectionName = tracker
+        self.attachDefaultSectionNameIfNeeded(&trackData)
```

Comment:
This appears to be a complicated setup that could be replaced with a simple if statement.  Something along the lines of 
```
if (tracker.sectionName == nil) { 
  tracker.sectionName = Constants.Track.defaultTrackingSectionName 
}
```